src/utils/embedded/embedded.provider.tsx

```diff
@@ -1214,6 +1249,69 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
       UNAUTHORIZED errors. When that happens, we should force de-authentication of the frontend.
     */
 
+      // Check for passkey auth data in localStorage immediately on startup
+      const customSessionId = localStorage.getItem("sessionId");
+      const customUserId = localStorage.getItem("userId");
+      const isCustomAuth = localStorage.getItem("isCustomAuth") === "true";
+
+      // If we have passkey auth data, initialize immediately without waiting
+      if (isCustomAuth && customSessionId && customUserId) {
+        console.log(
+          "Found passkey authentication data in localStorage, initializing immediately"
+        );
+
+        // Build a minimal session object for wallet activation
+        const dbSession = {
+          id: customSessionId,
+          userId: customUserId,
+          deviceNonce: localStorage.getItem("deviceNonce") || undefined,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          ip: "127.0.0.1",
+          userAgent: navigator.userAgent
+        };
+
+        // Create a custom auth token for API requests
+        const deviceNonce =
```

Comment:
You should use `getDeviceNonce` here, which takes care of loading/initializing/storing it.